clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -80,6 +91,287 @@ abstract class OpenstackClientProvider {
   }
 
   /**
+   * Gets load balancer pool for a given region by load balancer UUID.
+   * @param region
+   * @param loadBalancerId
+   * @return
+   */
+  LbPool getLoadBalancerPool(final String region, final String loadBalancerId) {
+    LbPool result = handleRequest {
+      getRegionClient(region).networking().loadbalancers().lbPool().get(loadBalancerId)
+    }
+    if (result == null) {
+      throw new OpenstackProviderException("Unable to find load balancer ${loadBalancerId} in ${region}")
+    }
+    result
+  }
+
+  /**
+   * Gets VIP for a given region.
+   * @param region
+   * @param vipId
+   * @return
+   */
+  Vip getVip(final String region, final String vipId) {
+    Vip result = handleRequest {
+      getRegionClient(region).networking().loadbalancers().vip().get(vipId)
+    }
+    if (result == null) {
+      throw new OpenstackProviderException("Unable to find vip ${vipId} in ${region}")
+    }
+    result
+  }
+
+  /**
+   * Validates the subnet in a region.
+   * @param region
+   * @param subnetId
+   * @return boolean
+   */
+  boolean validateSubnetId(final String region, final String subnetId) {
+    handleRequest {
+      getRegionClient(region).networking().subnet().get(subnetId) != null
+    }
+  }
+
+  /**
+   * Creates load balancer pool in provided region.
+   * @param region
+   * @param loadBalancerPool
+   * @return LbPool
+   */
+  LbPool createLoadBalancerPool(final String region, final LoadBalancerPool loadBalancerPool) {
+    handleRequest {
+      getRegionClient(region).networking().loadbalancers().lbPool().create(
+        Builders.lbPool()
+          .name(loadBalancerPool.derivedName)
+          .protocol(Protocol.forValue(loadBalancerPool.protocol?.name()))
+          .lbMethod(LbMethod.forValue(loadBalancerPool.method?.name()))
+          .subnetId(loadBalancerPool.subnetId)
+          .description(loadBalancerPool.description)
+          .adminStateUp(Boolean.TRUE)
```

Comment:
Just a programming style I guess ... Kind of equate it with HTTP Method's or status codes, prefer to use a constant.